or_issue.py

```diff
@@ -0,0 +1,210 @@
+except FileNotFoundError:
+    pass
+
+mkdirp(destination_folder)
+
+# Phase 3: Process TCL Scripts To Find Full List Of Files
+tcls_to_process = deque([ script_path ])
+def shift(deque):
+    try:
+        return deque.popleft()
+    except:
+        return None
+
+envs_used = ["OR_SCRIPT"]
+env["OR_SCRIPT"] = script_path_containerized
+
+current = shift(tcls_to_process)
+while current is not None:
+    script = open(current).read()
+
+    for key, value in env.items():
+        key_accessor = f"$::env({key})"
+        if not key_accessor in script:
+            continue
+        envs_used.append(key)
+        if value.endswith(".tcl"):
+            tcls_to_process.append(value)
+
+    current = shift(tcls_to_process)
+
+# Phase 4: Copy The Files
+final_env_pairs = []
+
+pdk_path = join(destination_folder, "pdk")
+openlane_misc_path = join(destination_folder, "openlane")
+
+def copy(frm, to):
+    parents = dirname(to)
+    mkdirp(parents)
+    try:
+        if isdir(frm):
+            shutil.copytree(frm, to)
+        else:
+            shutil.copyfile(frm, to)
+    except:
+        print(f"ℹ Couldn't copy {frm}, skipping...", file=sys.stderr)
+
+for key in envs_used:
+    value = env[key]
+    if value.startswith(run_path_containerized):
+        relative = relpath(value, run_path_containerized)
+        final_value = join(".", relative)
+        final_path = join(destination_folder, final_value)
+        from_path = value.replace(run_path_containerized, run_path)
+        copy(from_path, final_path)        
+        final_env_pairs.append((key, final_value))
+    elif value.startswith(pdk_root):
+        relative = relpath(value, pdk_root)
+        final_value = join("pdk", relative)
+        final_path = join(destination_folder, final_value)
+        copy(value, final_path)
+        final_env_pairs.append((key, final_value))
+    elif value.startswith("/openLANE_flow"):
+        relative = relpath(value, "/openLANE_flow")
+        final_value = join("openlane", relative)
+        final_path = join(destination_folder, final_value)
+        from_path = value.replace("/openLANE_flow", openlane_path)
+        copy(from_path, final_path)       
+        final_env_pairs.append((key, final_value))
+    else:
+        final_env_pairs.append((key, value))  
+
+# Phase 5: Create Run File
+run_cmd = join(destination_folder, "run")
+with open(run_cmd, "w") as f:
+    env_list = "\\\n    ".join([f"-e {key}='{value}'" for key, value in final_env_pairs])
+    f.write(f"""
+#!/bin/sh
+dir=$(cd -P -- "$(dirname -- "$0")" && pwd -P)
+cd $dir;
+docker run --rm\\
+    -tiv `pwd`:`pwd` -w `pwd`\\
+    {env_list}\\
+    {os.getenv("IMAGE_NAME") or "efabless/openlane:current"} openroad \\$::env\\(OR_SCRIPT\\)
+    """)
+os.chmod()
+
+# Phase 6: Tarball and output
+last_output = destination_folder
+if tarball:
+    last_output = destination_folder + ".tar.gz"
+    os.system(f"tar -cv {destination_folder} | xz | gzip > {last_output}")
```

Comment:
Why apply xz & gzip?  Also xz is quite slow so it would be good to make this an option.

Reply:
Also the name should be .tar.xz.gz

Reply:
I don't see any issue with uploading a .xz.  See https://github.com/maliberty/xz_test

Reply:
Sorry I missed that the issue is attaching to an issue and there the list is oddly limited.